runtime/Dict.res

```diff
@@ -1,4 +1,4 @@
-type t<'a> = Js.Dict.t<'a>
+type t<'a> = dict<'a>
```

Comment:
We should probably just get rid of this `type t` entirely and use `dict` directly.

Reply:
Yes, was thinking about that, too, as it is inconsistent with other Core modules (String, Array).

Should I do it in this PR or in a separate one?

Reply:
This one I think.